### DISCUSSION TITLE
6-2 Help on Connecting to the Database

### DISCUSSION THREAD

**Discussion by mhbangs-ua** [ua-gist604b-f24/syllabus#38] [General]
Hello,

I am trying to set up the connection to the OSM data and was able to set up the geoserver and load the osm layers to the server but I cant seem to view the layers or change the data store settings. I get this error 403 forbidden. I was following along in Professor Olssons video and he was able to just save and it worked but when I click save it does not save. Any ideas?

![image](https://github.com/user-attachments/assets/f8c8ab17-36f9-4af3-b088-24a7055cbd83)

![image](https://github.com/user-attachments/assets/f5f07363-aad1-48a2-b07f-59c7170f298a)

I did notice that the postgresqlexporer pane is blank so maybe the database is not actually up?
![image](https://github.com/user-attachments/assets/efb5878c-e220-4923-96d5-8d2061c0c8b4)

I know this is similar to Khartmans but it is slightly different I think?

**Comment by mhbangs-ua** [comment]
but the database already exists... 
![image](https://github.com/user-attachments/assets/4a9ffe5b-52eb-4b1a-8576-e39918887229)

**Reply by mhbangs-ua** [reply]
I dont know why my geoserver doesnt let me access certain areas also, I get this error 
![image](https://github.com/user-attachments/assets/cb952e7e-5b1d-4fc2-a887-6d5c2833abbc)

**Reply by mhbangs-ua** [reply]
also my fork is synced 
![image](https://github.com/user-attachments/assets/f1967995-b99c-4fcd-ac89-59facfac1938)

**Reply by aaryno** [reply]
The 403 Forbidden error is possibly a CORS error. This has plagued us in the past. It's a security thing where a webpage on one site can't include dynamic content from another site. The way geoserver constructs the links doesn't mesh well with how the browser understands urls and so some of the content is blocked. It's not an easy problem to solve. It only is blocked on the web interface, however. There is a REST url that will allow you to bypass that. https://docs.geoserver.org/latest/en/user/rest/styles.html.

1) Save your SLD file to your codespace. For this example, call it "my_style.sld"

2) From your terminal
```
curl -v -u admin:geoserver -XPOST -H "Content-type: text/xml" -d @my_style.sld http://localhost:8080/geoserver/rest/styles
```

We might need to do a little debugging but that's the gist of it. 

Some other thoughts...

postgres explorer is blank just because you haven't created a connection for that extension. not a problem.

**Reply by mhbangs-ua** [reply]
I tried this a few different ways but I dont think that command worked. I got a long response that I could not copy and paste from the terminal but here are a few screenshots
![image](https://github.com/user-attachments/assets/cb916076-2c67-4402-9d35-bd7b77f2ef43)
![image](https://github.com/user-attachments/assets/c054990c-7382-4d77-a1b3-5ce270922ee8)

I am going to keep looking into this

**Reply by mhbangs-ua** [reply]
which SLD file am I supposed to be saving? Wouldnt there be one for each layer? 
![image](https://github.com/user-attachments/assets/2507137f-72c0-42e8-8dc1-d3a6856f965d)

**Reply by mhbangs-ua** [reply]
![image](https://github.com/user-attachments/assets/3333b3dc-92cd-4100-82f6-1fc2a6602e18)
![image](https://github.com/user-attachments/assets/506755a1-4090-43d2-aa8d-7bfc3a6c093b)

**Reply by mhbangs-ua** [reply]
success? But that doesnt fix anything

**Comment by travisz09** [comment]
@mhbangs-ua, it sounds like you are having the same issue that Jen experienced. If you are still stuck on this assignment, then see our discussion on [https://github.com/ua-gist604b-f24/syllabus/discussions/35](https://github.com/ua-gist604b-f24/syllabus/discussions/35).

**Reply by mhbangs-ua** [reply]
I really appreciate the insight Travis, was hoping that the professor would help out on this one though. Kind of disappointed after spending hours trying to recreate this assignment on my own.
